packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -113,6 +138,25 @@ export default class Menu extends Component {
     return signOut()
   }
 
+  _subscribeMissingPatches = () => {
+    const unsubs = map(this.props.hosts, host =>
+      subscribeHostMissingPatches(host, patches => {
+        this.setState({
+          missingPatches: {
+            ...this.state.missingPatches,
+            [host.id]: patches.length,
```

Comment:
`[host.id]: patches.length > 0`: you only need to know if there are some missing patches.